src/sentry/api/endpoints/sentry_internal_app_token_details.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+from rest_framework import status
+from django.http import Http404
+
+from sentry.api.bases import (
+    SentryInternalAppTokenPermission, SentryAppBaseEndpoint,
+)
+from sentry.models import ApiToken
+from sentry.features.helpers import requires_feature
+from sentry.mediators.sentry_app_installation_tokens import Destroyer
+
+
+class SentryInternalAppTokenDetailsEndpoint(SentryAppBaseEndpoint):
+    permission_classes = (SentryInternalAppTokenPermission, )
+
+    def convert_args(self, request, sentry_app_slug, token, *args, **kwargs):
+        # get the sentry_app from the SentryInternalAppTokenDetailsEndpoint class
+        (args, kwargs) = super(SentryInternalAppTokenDetailsEndpoint,
+                               self).convert_args(request, sentry_app_slug, *args, **kwargs)
+
+        try:
+            kwargs['api_token'] = ApiToken.objects.get(token=token)
+        except ApiToken.DoesNotExist:
+            raise Http404
+
+        return (args, kwargs)
+
+    @requires_feature('organizations:sentry-apps', any_org=True)
+    def delete(self, request, sentry_app, api_token):
+        if not sentry_app.is_internal:
+            return Response('This route is limited to internal integrations only',
+                            status=status.HTTP_403_FORBIDDEN
+                            )
+
+        # Validate the token is associated with the application
+        if api_token.application_id != sentry_app.application_id:
```

Comment:
I would maybe put this check before the `is_internal` check since it seems like someone could then know someone else app is internal

Reply:
@MeredithAnya We check the permissions of the sentry_app before we get here so you would only know if the app is internal if you have permissions for it. See `convert_args` of `SentryAppBaseEndpoint`.

Reply:
ahh okays got it. think i misread your response to mark above. we do this validation so that you don't accidentally delete a token for another one of your sentry apps.

Reply:
Correct

Reply:
@MeredithAnya I ended up flipping the checks because I want to always return a `404` if the token doesn't exist for that sentry app.